### PR TITLE
grandpa: save historical votes

### PR DESCRIPTION
**Pull request by marcio-diaz**
We need to save the votes received indicating the moment of voting in order to answer challenge queries.

Related to https://github.com/paritytech/finality-grandpa/issues/10

Depends on: https://github.com/paritytech/finality-grandpa/pull/65

### CODE REVIEW

**Comment by DemiMarie-temp** [comment]
The compile errors need to be fixed.

**Comment by bkchr** [comment]
What is the status of this?

**Comment by andresilva** [comment]
This is dependent on us releasing finality-grandpa 0.8, and we're trying to synchronize a couple of PRs that will depend on it (e.g. #2801). IMO this PR and #2611 could be merged into one.

File: core/finality-grandpa/src/aux_schema.rs.1 [hunk 1]
```diff
@@ -0,0 +1,938 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
```
Comment (andresilva) [inline core/finality-grandpa/src/aux_schema.rs.1]: Committed by mistake?

File: core/finality-grandpa/src/environment.rs [hunk 2]
```diff
@@ -108,16 +109,13 @@ impl<Block: BlockT> Decode for CompletedRounds<Block> {
...
-		voters: &AuthoritySet<Block::Hash, NumberFor<Block>>,
+		voters: Vec<AuthorityId>,
 	)
 		-> CompletedRounds<Block>
 	{
-		let mut rounds = VecDeque::with_capacity(NUM_LAST_COMPLETED_ROUNDS);
-		rounds.push_back(genesis);
-
-		let voters = voters.current().1.iter().map(|(a, _)| a.clone()).collect();
+		// let voters = voters.current().1.iter().map(|(a, _)| a.clone()).collect();
```
Comment (andresilva) [inline core/finality-grandpa/src/environment.rs]: Remove?

**Comment by bkchr** [comment]
What is the status of this?

**Comment by marcio-diaz** [comment]
> What is the status of this?

After a talk with a w3f guy we realized that is not needed.

**Pull request closed without merging**